agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1806,10 +1800,44 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
     } break;
     case Event::STA_Connected: {
         LOG(TRACE) << "Received STA_Connected event";
+        auto msg = static_cast<bwl::sACTION_MONITOR_CLIENT_ASSOCIATED_NOTIFICATION *>(data);
+
+        auto sta_mac = tlvf::mac_to_string(msg->mac);
+        auto vap_id  = msg->vap_id;
+
+        LOG(INFO) << "STA_Connected: mac=" << sta_mac << " vap_id=" << vap_id;
+
+        mon_db.sta_erase(sta_mac);
```

Comment:
Why do we need to erase the STA anyway?
If it's already in the DB, why not simply move it to the correct vap?
This logic can be added to `sta_add` method.

Reply:
When we erase and add it again, all node class members are set to default values.
This is the only thing I could think of.

Reply:
Yeah, I think it's better to keep that behaviour.